### DISCUSSION TITLE
How to access cookies in this case?

### DISCUSSION THREAD

**Discussion by mblachlov24** [nuxt/nuxt#16254] [Questions]
composables/useApi.ts
```js
export const useApi = $fetch.create({
  baseURL: 'xxxx',
  credentials: 'include',
  headers: {
    'X-XSRF-TOKEN': // useCookie not working in this place (nuxt instance unavailable)
  }
})
```

![image](https://user-images.githubusercontent.com/109919875/204276111-a5e1d35e-a5bf-4f64-a3fc-090803746481.png)

I need to export it like this so that when invoked I can set typescript types.

**Comment by rafifn** [comment]
how about `getCookie` from [h3](https://github.com/unjs/h3#built-in) ?

updated:
oopss sorry, `getCookie` require event

**Reply by rafifn** [reply]
works fine for me
```
export const useApi = $fetch.create({
  baseURL: 'xxxx',
  credentials: 'include',
  headers: {
    'X-XSRF-TOKEN': useCookie('token')?.value ?? '',
  },
})

useApi('/halo')
```

**Reply by mblachlov24** [reply]
Not working :(

```
500
nuxt instance unavailable

at Module.useRequestEvent (./node_modules/nuxt/dist/app/composables/ssr.mjs:14:58)
at readRawCookies (./node_modules/nuxt/dist/app/composables/cookie.mjs:46:62)
at Module.useCookie (./node_modules/nuxt/dist/app/composables/cookie.mjs:22:19)
at ./composables/useApi.ts:7:43
at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
at async ViteNodeRunner.directRequest (./node_modules/vite-node/dist/client.mjs:270:5)
at async ViteNodeRunner.cachedRequest (./node_modules/vite-node/dist/client.mjs:146:12)
at async request (./node_modules/vite-node/dist/client.mjs:169:16)
at async ./pages/index.js:1:31
```

**Reply by TheAlexLichter** [reply]
Could you provide a reproduction? 🤔

**Reply by mblachlov24** [reply]
@manniL Please, here's my reproduction: https://stackblitz.com/edit/nuxt-starter-oh6uab?file=composables/useApi.ts

**Comment by TheAlexLichter** [comment]
[Initial StackBlitz from @mblachlov24](https://stackblitz.com/edit/nuxt-starter-oh6uab?file=composables/useApi.ts)

[Solution StackBlitz](https://stackblitz.com/edit/nuxtframeworkdiscussions9376?file=composables%2FuseApi.ts,app.vue) (naive, can be nicer)

Answer: You have to wrap the `$fetch.create` call in a function. Else, it will be executed during the *import* and not during the `setup` function. Thus, there won't be any nuxt instance.

**Comment by mblachlov24** [comment]
Thanks for answer. I think I found a better way. What do you think about it?

```js
export const useApi = $fetch.create({
  baseURL: 'xxx',
  credentials: 'include',
  onRequest({ options }) {
    options.headers = {
      'X-XSRF-TOKEN': useCookie('XSRF-TOKEN')?.value ?? ''
    }
  }
})
```
This allows to simply use:
```js
const test = await useApi('/test')
```
